### DISCUSSION TITLE
Gizmo lights not right, messed up firmware?

### DISCUSSION THREAD

**Discussion by CrazyCatRobotics** [gizmo-platform/gizmo#30] [Q&A]
Hello,
We believed that our gizmo had no firmware loaded on it, so we downloaded firmware from GitHub detailed in the Advanced_Team_Topics_v6.pdf presentation to the gizmo. Now the network light is red, the field light is off (no firmware?), and the battery light is green. We cannot figure out how to make the lights turn their correct colors.

**Comment by Vaelatern** [comment]
Have you tried to pair this with your driver station?

**Reply by CrazyCatRobotics** [reply]
Yes, we paired it and it was successful. We messed with the firmware after pairing it, then paired it successfully again.

**Reply by barulicm** [reply]
The light pattern described (red, off, green) is what you see if the Gizmo has been bound but can't find the network connection to the driver station.
One thing you can try is to connect your driver station and Gizmo with the tether cable. This will bypass anything that might be wrong with the wireless connection. If you're sure the driver station is on (you should see an orange and green light next to the tether port on the driver station), this pattern could mean that the Gizmo has an incorrect binding or the driver station is having some sort of trouble.
You can find a diagram of the most common Gizmo status light patterns here: https://gizmoplatform.dev/trouble/gizmo.html#dynamic-status-indicators

**Reply by CrazyCatRobotics** [reply]
We still have the same issue, and correctly bound the driver's station. It is connected by the tether cable. The issue must be with the firmware, as we downloaded multiple copies of it onto the Gizmo by accident. Sometimes the Network and Field lights are both flashing red, and sometimes the Network light is red while the Field light is off. How can we fix it? Should we clear the files on the Gizmo? If so, how?

**Reply by the-maldridge** [reply]
It is not possible to download multiple copies of the firmware concurrently, each successive copy fully replaces the previous one.  If you're seeing the field/network double red blinking that's a pairing fault and you should re-pair.

All the other sequences of lights you have described are standard positions within the state machine while the system acquires the control stream.  The link to the troubleshooting guide explains these states.

**Reply by Vinnierxy** [reply]
Hi there,

Our team is currently having the same issue. We're trying to see if we can solve it by resetting the processor and downloading the firmware from GitHub, but using the information on the same presentation (Advanced_Team_Topics_v6.pdf), we aren't sure how to proceed. It says to download the latest release of the gss uf2 file, but the GitHub shows 4 differently labeled files all released on the same day with the same size and everything. Which file is the correct one? Do we download all of these? If so, would we not run into the issue of wiping the data from the previous file as we dialed the next?

**Reply by Vaelatern** [reply]
@Vinnierxy this is an Advanced Team Topic.

You will need the firmware that matches your board revision.

But back up.

What are your lights?

What are they closest to in the doc?

Why is the light pattern as documented not valid in your circumstance?
